### DISCUSSION TITLE
HMIP-RFUSB wird unter CasaOS (Docker) auf Raspberry 4 (8GB) nicht erkannt

### DISCUSSION THREAD

**Discussion by derAuge** [OpenCCU/OpenCCU#2477] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Der USB-Stick (HMIP-RFUSB (4.4.18)) wird nicht von RaspberryMatic im DockerContainer unter CasaOS erkannt.

`ccu  | Updating Homematic RF-Hardware: HMIP-RFUSB: =>4.4.18, ERROR ( != 4.4.18)`

Den Stick habe ich in einem RPi3 mit RaspberryMatic getestet. dort wurde er erkannt:

```
homematicIP-RF (HmIP) Info
RF-Module/Firmware:	HMIP-RFUSB (4.4.18)
Device-Node:	/dev/raw-uart1 (eQ-3 HmIP-RFUSB@usb-3f980000.usb-1.2)
Address:	0xBAE3D1 (0xBAE3D1)
SGTIN:	xxxxxxxx00420C9A17DD0
Serial:	xxxxxxx
DC / CS:	1 % / 0 %
HomeMatic-RF (HmRF) Info
RF-Module/Firmware:	HMIP-RFUSB (4.4.18)
Device-Node:	/dev/raw-uart1 (eQ-3 HmIP-RFUSB@usb-3f980000.usb-1.2)
Address:	0xFF99A1 (0xFF99A1)
Serial:	xxxxxxx
DC / CS:	1 % / 0 %
```

### Describe the behavior you expected

auf der Konsole des RPi4:

```
auge@RPi4-1:~ $ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 005: ID 0403:6001 Future Technology Devices International, Ltd FT232 Serial (UART) IC
**Bus 001 Device 004: ID 1b1f:c020 eQ-3 Entwicklung GmbH HmIP-RFUSB**
Bus 001 Device 003: ID 046a:00a1 CHERRY SmartCard Reader Keyboard KC 1000 SC
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```
```
auge@RPi4-1:~ $ lsusb -v -s 001:004

Bus 001 Device 004: ID 1b1f:c020 eQ-3 Entwicklung GmbH HmIP-RFUSB
Couldn't open device, some information will be missing
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               1.10
  bDeviceClass            0
  bDeviceSubClass         0
  bDeviceProtocol         0
  bMaxPacketSize0        64
  idVendor           0x1b1f eQ-3 Entwicklung GmbH
  idProduct          0xc020 HmIP-RFUSB
  bcdDevice            1.00
  iManufacturer           1 Silicon Labs
  iProduct                2 eQ-3 HmIP-RFUSB
  iSerial                 3 xxxxxxxxA0000420C9A17DD0
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0020
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0
    bmAttributes         0x80
      (Bus Powered)
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass       255 Vendor Specific Class
      bInterfaceSubClass      0
      bInterfaceProtocol      0
      iInterface              2
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               0
auge@RPi4-1:~ $
```

### Steps to reproduce the issue

1. Konsole RPi4 = Bus 001 Device 004: ID 1b1f:c020 eQ-3 Entwicklung GmbH HmIP-RFUSB
2. Konsole Container = Bus 001 Device 004: ID 1b1f:c020
3. Es wird weder auf dem RPi4 noch im Container ein ttyUSBx angelegt
4. Im Container:
```
# cat /var/hm_mode 
HM_HMIP_ADDRESS=''
HM_HMIP_ADDRESS_ACTIVE=''
HM_HMIP_DEV=''
HM_HMIP_DEVNODE=''
HM_HMIP_DEVTYPE=''
HM_HMIP_SERIAL=''
HM_HMIP_SGTIN=''
HM_HMIP_VERSION=''
HM_HMRF_ADDRESS=''
HM_HMRF_ADDRESS_ACTIVE=''
HM_HMRF_DEV=''
HM_HMRF_DEVNODE=''
HM_HMRF_DEVTYPE=''
HM_HMRF_SERIAL=''
HM_HMRF_VERSION=''
HM_HOST='oci'
HM_LED_GREEN=''
HM_LED_GREEN_MODE1='mmc0'
HM_LED_GREEN_MODE2='heartbeat'
HM_LED_RED=''
HM_LED_RED_MODE1='timer'
HM_LED_RED_MODE2='mmc0'
HM_LED_YELLOW=''
HM_LED_YELLOW_MODE1='none'
HM_LED_YELLOW_MODE2='none'
HM_MODE='NORMAL'
HM_RTC=''
```

### What is the version this bug report is based on?

3.71.12.20231014 (oci)

### Which base platform are you running?

rpi4 (RaspberryPi4)

### Which HomeMatic/homematicIP radio module are you using?

HmIP-RFUSB

### Anything in the logs that might be useful for us?

```txt
Meine Fragen:
1. warum wird der Stick im RPi4 nicht als ttyUSBx angelegt?
2. Dadurch kann ein "durchreichen" in den Container nicht eingerichtet werden
3. Warum wird der Stick trotzdem erkannt (mit lsusb)
```

### Additional information

**Zum System:**
auf dem RPi4 (8GB) läuft das "Raspberry Pi OS Lite (64-bit)" Image
Darauf wurde "CasaOS" installiert (curl -fsSL https://get.casaos.io | sudo bash)
Dort wurden dann die Container installiert:
- Grafana
- InfluxDB
- ioBroker
- Solaranzeige
- RaspbarryMartic

**Comment by jens-maus** [comment, marked as answer]
Offensichtlich wurde hier nicht anhand der offiziellen Anleitung vorgegangen unter zuhilfenahme des `install-docker.sh` skriptes (https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI). Kurz zusammengefasst: Die HomeMatic Funkmodule brauchen spezielle Kernelmodule die auf dem Docker-Host System installiert und entsprechend permissions für die Docker Container eingerichtet sein müssen. Das ganze ist nicht trivial weil eben eine Hardwareabhängig zu einem Kerneltreiber gegeben ist.

Bitte also erst einmal in einem Diskussionsforum Hilfe suchen bzw. dein Einsatzzweck diskutieren denn das ist erst einmal kein Bug, insofern ist das hier nicht wirklich richtig, denn das hier ist ein Bug/Issue-Ticket.

**Comment by derAuge** [comment]
Danke für die schnelle Rückmeldung.
Leider bin ich davon Ausgegangen das bei einer Docker-Installation alles notwendigen Pakete mit eingerichtet werden.  Habe gelernt das es bei RaspberryMatic anders ist. Hier muss der Host angepasst werden. = Okay

Hier meiner (von euch vorgeschlagenen) Anpassungen die ich auf dem RPi4 (Host) installiert habe.
1.)
```
sudo apt install wget ca-certificates build-essential bison flex libssl-dev gpg
wget -qO - https://apt.pivccu.de/piVCCU/public.key | sudo gpg --dearmor -o /usr/share/keyrings/pivccu-archive-keyring.gpg
sudo sh -c 'echo "deb [signed-by=/usr/share/keyrings/pivccu-archive-keyring.gpg] https://apt.pivccu.de/piVCCU stable main" >/etc/apt/sources.list.d/pivccu.list'
sudo apt update
sudo apt install pivccu-modules-dkms
```
2.)
`sudo apt install pivccu-modules-raspberrypi`

3.)
```
sudo bash -c 'cat <<EOF >/etc/udev/rules.d/99-Homematic.rules
ATTRS{idVendor}=="1b1f" ATTRS{idProduct}=="c020", ENV{ID_MM_DEVICE_IGNORE}="1"
ATTRS{idVendor}=="1b1f" ATTRS{idProduct}=="c00f", ENV{ID_MM_DEVICE_IGNORE}="1"
ATTRS{idVendor}=="0403" ATTRS{idProduct}=="6f70", ENV{ID_MM_DEVICE_IGNORE}="1"
ATTRS{idVendor}=="10c4" ATTRS{idProduct}=="8c07", ENV{ID_MM_DEVICE_IGNORE}="1"
EOF'

```
4.)
`sudo sh -c 'echo eq3_char_loop >/etc/modules-load.d/eq3_char_loop.conf'`


5.)
```
sudo service pivccu-dkms start
sudo modprobe eq3_char_loop
```


Das alles konnte ich Dank der [Anleitung ](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI#manual-installation)durchführen. Danach wurde der Stick erkannt und ich konnte meine Rauchmelder anlernen

Noch einmal Danke für die Bereitstellung und den schnellen Support !

Gruß
Jürgen

**Reply by jens-maus** [reply]
Und warum hast du nicht das vorgeschlagene `install-docker.sh` install script bzw. den [Automatic Installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI#automatic-installation) teil verwendet? Weil das ist im Grunde die vorgeschlagene Installationsmethode.
